### PR TITLE
[3832] disable option when out of stock

### PR DESCRIPTION
**Pull request by pablobentivengo**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3832

**Problem:**
* Bundle Dropdown options were able to select even when the product was out of stock

**In this PR:**
* Modified Util/Product/Transform file,  bundleOptionsToSelectTransform function. Added the isAvailable parameter to option.

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
Issue 1: Check box for 'out of stock' option is not shaded and disabled
Please, make name of option grey too, same as in drop down
image
Issue 2: if user set child product as selected by default in admin and it is 'out of stock' than product should be disabled too
Bundle product: https://pr-4110.d25kd40decjpue.amplifyapp.com/default/8bun.html
Out of stock option: 8 simple
https://vimeo.com/667811203/b11ecfeadd

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: all check boxes marked as selected when user open product and can not unselected (same for any other checkboxes, for example PLP)
https://vimeo.com/668164121/98e08cfe25
Only selected by default in admin option should be marked as selected when user open PDP
Also user should be able to remove default selection

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: fixed
Issue 4: 
Line is missing after the 'In stock' option in dropdown
image
Issue 5: 
'Out of stock' Radio button option is not disabled for bundle product
Product: https://pr-4110.d25kd40decjpue.amplifyapp.com/default/bundl-with-out-of-stock-radio-buttons.html
image
Expected result:
image

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: fixed
Issue 4: failed: IOS - ok, Android - lines absent
image
Issue 5: fixed

**Comment by pablobentivengo** [comment]
Hey Elena, after too many failed attempts to solve the issue with @AleksandrsKondratjevs, we got to a documentation where it says that it is not possible to override Option Element as the styling for these element is handled by the user's OS. 

As the documentation says, it is only possible to style the background-color and color properties. We tried to style to find a possible solution, but we could not do it. 

You can check the documentation we found [here].(https://web.archive.org/web/20170528070441/http://msdn.microsoft.com/en-us/library/ms535877(v=vs.85).aspx)

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: fixed
Issue 4: fixed
Issue 5: fixed
Issue 6: Grey field overlap the button 'Add to cart'

**Comment by Elena-Evsa** [comment]
Issue 1: fixed
Issue 2: fixed
Issue 3: fixed
Issue 4: fixed
Issue 5: fixed
Issue 6: fixed
Issue 7: disable hover affect for out of stock option in checkboxes and radio buttons
https://vimeo.com/673077519/aeeec2fd37

File: packages/scandipwa/src/component/Field/Field.component.js [hunk 1]
```diff
@@ -203,16 +203,18 @@ export class Field extends PureComponent {
             onChange: onChange || noopFn
         };
 
+        const disableCheckbox = isDisabled ? !isDisabled : null;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Field/Field.component.js:206]: Won't just putting isDisabled work for checked attribute ?

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Field/Field.component.js (hunk 1 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/Field/Field.component.js:206]: If I leave isDisabled, default checked option wouldn't be checked and stay unchecked even when trying to select them.
Like in this video https://vimeo.com/674792314/a12b76c701.

If left as is: https://vimeo.com/674792819/f61fc2b5db

File: packages/scandipwa/src/component/Field/Field.component.js [hunk 2]
```diff
@@ -203,16 +203,18 @@ export class Field extends PureComponent {
...
         return (
-            <label htmlFor={ id } block="Field" elem={ `${elem}Label` }>
+            <label htmlFor={ id } block="Field" elem={ `${elem}Label` } mods={ { isDisabled } }>
                 <input
                   ref={ (elem) => setRef(elem) }
                   disabled={ isDisabled }
                   type={ type }
                   { ...attr }
                   { ...inputEvents }
+                  checked={ disableCheckbox }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Field/Field.component.js:215]: isDisabled ? !isDisabled : null; move it here

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**